### ISSUE TITLE
Router.push CORS headers error

### ISSUE THREAD

**Issue by talentlessguy** [zeit/next.js#10558]
# Bug report

When I try to make a redirect with `Router.push` to a page, it grabs a header from the current page, uses it in the  redirect and it results in CORS request.

## Describe the bug

Instead of making a redirect I get this error:

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://komfy.now.sh/verify_email. (Reason: header ‘x-captcha’ is not allowed according to header ‘Access-Control-Allow-Headers’ from CORS preflight response).

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://komfy.now.sh/verify_email. (Reason: CORS request did not succeed).
```

## To Reproduce

Here are the links to a repository, where the problem happens, and a live demo

[repo](https://github.com/komfy/frontend) | [piece of code where redirect happens](https://github.com/komfy/frontend/blob/b1fd3de254e0cc25cf3d63acf040b1c6b3e48f34/pages/register.tsx#L51) | [live demo](https://komfy.now.sh/register)

## Expected behavior

Redirect should happen without any errors.

## System information

- OS: Manjaro Linux x64
- Browser (if applies) Firefox 70
- Version of Next.js: 9.2.1

**Comment by lfades** [comment]
Hi, your code looks fine to me, I have a couple questions:
- Does it work locally?
- Do you get the same error in another browser, like Chrome, or only in Firefox?

**Comment by talentlessguy** [comment]
@lfades 

* I don't know how but it still requests another host, even on local thing. This is verrryy weird.

![](https://i.snipboard.io/oSGgs5.jpg)

* I get this error in Brave (Chrome-based browser) not on localhost:

```
Access to fetch at 'https://komfy.now.sh/verify_email' (redirected from 'https://api.komfy.now.sh/reg') from origin 'null' has been blocked by CORS policy: Request header field x-captcha is not allowed by Access-Control-Allow-Headers in preflight response.
```

For some reason origin is `null` :thinking: 

### Workaround

I can replace `Router.push` with `location?.href = '/verify_email'` but there won't be client-side transition anymore.

**Comment by lfades** [comment]
![image](https://user-images.githubusercontent.com/4278345/74683197-22434680-5196-11ea-9c86-af17daa45469.png)

That looks more like an error with the request to your API than an error with Next.js, please make sure that you're making a request to your API with CORS enabled.

**Comment by talentlessguy** [comment]
@lfades oh yes, looks like it is API's fault. Closing the issue!

![image](https://user-images.githubusercontent.com/35937217/74685186-d072de80-51de-11ea-9450-927ec586c41c.png)

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
